src/Doctrine/ORM/InvoiceRepository.php

```diff
@@ -29,13 +30,17 @@ public function findOneByOrder(OrderInterface $order): ?InvoiceInterface
 
     public function findByOrderNumber(string $orderNumber): array
     {
-        return $this
+        $orders = $this
             ->createQueryBuilder('invoice')
             ->innerJoin('invoice.order', 'o')
             ->where('o.number = :orderNumber')
             ->setParameter('orderNumber', $orderNumber)
             ->getQuery()
             ->getResult()
         ;
+
+        Assert::isArray($orders, 'Result of query should be an array.');
```

Comment:
```suggestion
        Assert::isArray($invoices);
```
No point in overwriting the default message.